### PR TITLE
smart disclaimer - persist close

### LINKED ISSUE #535: Allow to close the disclamer
Persist the closing of the disclaimer

### Estimation:
*0.5*

### PR DESCRIPTION
**Pull request by W3stside**
Closes #535 

As title states - very simple implementation

I think in the future, with more persisted state, we should consider using [`localForage`](https://github.com/localForage/localForage) which allows for more than just strings

Using JSON parse/stringify is a pain

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr577--dexreact.review.gnosisdev.com

**Comment by alfetopito** [comment]
Dumb question: I don't see the banner. Is it because it's not part of the new design yet?

File: src/components/LegalBanner.tsx [hunk 1]
```diff
@@ -72,17 +82,24 @@ const LegalText: React.FC = () => (
   </Text>
 )
 
-const LegalBanner: React.FC<LegalBannerProps> = ({ startOpen, useFull, title }) => {
-  const [open, setOpen] = useState(startOpen)
+const DISCLAIMER = 'GP_DISCLAIMER'
+
+const LegalBanner: React.FC<LegalBannerProps> = ({ useFull, title }) => {
+  const [open, setOpen] = useState(JSON.parse(localStorage.getItem(DISCLAIMER) as string))
```
Comment (Velenir) [inline src/components/LegalBanner.tsx]: It starts closed, as `localStorage.getItem(DISCLAIMER) === null` when nothing is yet in localStorage

Comment (W3stside) [reply src/components/LegalBanner.tsx]: true
good point
oops

Comment (W3stside) [reply src/components/LegalBanner.tsx]: i'll rock with strings

**Comment by W3stside** [comment]
thanks @Velenir good catch

File: src/components/LegalBanner.tsx [hunk 2]
```diff
@@ -72,17 +82,24 @@ const LegalText: React.FC = () => (
   </Text>
 )
 
-const LegalBanner: React.FC<LegalBannerProps> = ({ startOpen, useFull, title }) => {
-  const [open, setOpen] = useState(startOpen)
+const DISCLAIMER = 'GP_DISCLAIMER'
```
Comment (anxolin) [inline src/components/LegalBanner.tsx]: I think `DISCLAIMER` is too general. 

nit: Maybe `IS_CLOSED_PARAM` or sth like that.

File: src/components/LegalBanner.tsx [hunk 3]
```diff
@@ -72,17 +82,24 @@ const LegalText: React.FC = () => (
...
+
+const LegalBanner: React.FC<LegalBannerProps> = ({ useFull, title }) => {
+  const [open, setOpen] = useState(localStorage.getItem(DISCLAIMER) || 'OPEN')
+
+  useEffect(() => {
+    open === 'OPEN' ? localStorage.setItem(DISCLAIMER, 'OPEN') : localStorage.setItem(DISCLAIMER, 'CLOSED')
```
Comment (anxolin) [inline src/components/LegalBanner.tsx]: Being two the values, why not a boolean as a string?

```js
localStorage.setItem(IS_CLOSED_PARAM, 'true')
...
const [isClosed, setIsClosed] = useState(localStorage.getItem(IS_CLOSED_PARAM) === 'true')
...

```

File: src/components/LegalBanner.tsx (hunk 2 above)
Comment (W3stside) [reply src/components/LegalBanner.tsx]: it's the key name

File: src/components/LegalBanner.tsx (hunk 3 above)
Comment (W3stside) [reply src/components/LegalBanner.tsx]: good rec, will work it in

Comment (W3stside) [reply src/components/LegalBanner.tsx]: actually there are issues with the null type and the way it's stringified so for simplicities sake and the fact its working now I'll leave it as is :D if it's super annoying we can change it later

Comment (anxolin) [reply src/components/LegalBanner.tsx]: But `localStorage.getItem(IS_CLOSED_PARAM) === 'true'` 

As I understand, `localStorage.getItem(IS_CLOSED_PARAM)` can be:
* `null`: `null == 'true'`   ---> `false` (no closed)
* `'true'`: `'true' == 'true'`   ---> `true` (closed)

File: src/components/LegalBanner.tsx (hunk 2 above)
Comment (anxolin) [reply src/components/LegalBanner.tsx]: That's my point. `DISCLAIMER` var should make a reference of what it is. I think it should have then the prefix/suffix `_KEY`

**Pull request merged**